backend/src/nodes/ncnn_nodes.py

```diff
@@ -199,9 +154,10 @@ def upscale(
             raise RuntimeError("An unexpected error occurred during NCNN processing.")
 
     def run(
-        self, net_data: NcnnNetData, img: np.ndarray, tile_size_target: int
+        self, model: NcnnModel, img: np.ndarray, tile_size_target: int
     ) -> np.ndarray:
         h, w, _ = get_h_w_c(img)
+        model_c = self.get_model_channels(model)
```

Comment:
In the below code, you assume that `model_c` is either 1 or 3. Could you please add an assertion for this?